src/BinomialQueue/Internals.hs

```diff
@@ -209,6 +210,13 @@ singleton x = MinQueue (Cons (tip x) Nil)
 insert :: Ord a => a -> MinQueue a -> MinQueue a
 insert x (MinQueue ts) = MinQueue (incr (tip x) ts)
 
+-- | \(O(\log n)\), but a fast \(O(1)\) average when inserting repeatedly in
+-- an empty queue or at least around \(O(\log n)\) times into a nonempty one.
+-- Insert an element into the priority queue. This is good for 'fromList'-like
+-- operations.
```

Comment:
Should we also use it in `fromList`-like functions then? Or do you want to do that in a future PR?

Reply:
We currently tend to inline it, but yes, I'd like to hunt those down.

Reply:
Done.